src/engine/Combat.java

```diff
@@ -14,46 +13,59 @@
 	
 	private static Logger LOG = new Logger("Combat");
  	
-	private Grid[][] gameBoard;
+	private Board board;
 
-	public Combat(Grid[][] board) {
+	public Combat(Board board) {
 		
-		this.gameBoard = board; 
+		this.board = board; 
 	}
 	
 	/**
-	 * Combat calculations for a plant attacking a zombie. If a zombie was killed by this attack,
-	 * the coordinates of the zombie are retrieved to allow game to remove zombie from the game.
+	 * Performs a plant attack based on the type of plant.
 	 * 
 	 * @param source Plant that is attacking
-	 * @return coordinates of the zombie that was killed, null otherwise
 	 */
-	public int[] plantAttack(Plant source) {
+	public void plantAttack(Plant source) {
 		
-		int plantRow = source.getRow();
-		int damage = source.getPower();
+		switch (source.getPlantType()) {
+			case PEASHOOTER:
+				peashooterAttack((Peashooter) source);
+				break;
+			default:
+				break;
+		}
+	}
+	
+	/**
+	 * Perform a Peashooter attack. Find and damage the first zombie
+	 * target.
+	 * 
+	 * @param plant The peashooter Plant object
+	 */
+	public void peashooterAttack(Peashooter plant) {

```

Comment:
Maybe instead of peashooterAttack (attack methods per unit), we have different attack types like: 

Attack Single
Attack Grid
Attack Row
Attack Air

and we separate the attack logic that way? Each plant can have an attack type in its class.

Reply:
The reason why I didn't do general attack methods since some plant attacks have different effects. So for example, a snow peashooter can slow down units. Though `Attack Single, Attack Grid etc..` could be used to re-use code but I will still have to have separate methods.

Reply:
The other suggestion I could do is to remove  `Combat Engine` all together. Instead, let each Plant have their own `attack()` method. The downside is that Plants will have access to the Board.

Reply:
yeah we will need different methods for different attack types, but doing it for units seems too far